src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -56,13 +57,19 @@ sortObjects(Kokkos::View<unsigned int *, DeviceType> view)
   // better choice here because its size is guaranteed to coincide with the
   // pointer size which is a good thing for converting with reinterpret_cast
   // (when leaf indices are encoded into the pointer to one of their children)
-  Kokkos::BinSort<ViewType, CompType, DeviceType, size_t> bin_sort(
+  Kokkos::BinSort<ViewType, CompType, DeviceType, unsigned int> bin_sort(
       view, CompType(n / 2, result.min_val, result.max_val), true);
   bin_sort.create_permute_vector();
   bin_sort.sort(view);
   ExecutionSpace().fence();
 
-  return bin_sort.get_permute_vector();
+  auto permute_vector = bin_sort.get_permute_vector();
+  Kokkos::parallel_for(
+      ARBORX_MARK_REGION("copy_permute_vector"),
+      Kokkos::RangePolicy<ExecutionSpace>(0, n),
+      KOKKOS_LAMBDA(int i) { permute(i) = permute_vector(i); });
```

Comment:
This is silly now change the return type of the function and skip the copy.

Reply:
That was my first thought too, but we have to make permute indices compatible with `Node*` for storing, so no cigar.

Reply:
Remove https://github.com/arborx/ArborX/blob/dcc40adcc63f6bc253ec31f6050dd969f6e366c2/src/details/ArborX_DetailsTreeConstruction.hpp#L322-L325

I was planning on dropping that static assertion soon anyway.

Reply:
Not sure about this. It's already a potentially wrong thing to do to work with int. With your proposed change this will spread through the code.